index.html

```diff
@@ -807,6 +807,48 @@ <h3>Proof Verification (eddsa-2022)</h3>
         </section>
       </section>
 
+      <section>
+        <h3>json-eddsa-2022</h3>
```

Comment:
I wonder if this should be `jcs-eddsa-2022` instead. Otherwise, the rest of this is a good start, IMO.

Reply:
Yes, I had started w/ that and then became concerned that most developers won't know or care about what "JCS" is... even "eddsa" is a stretch -- most developers don't know anything about the details of how "EdDSA" works, or what parameters they should be using for their application... For this particular cryptosuite, they're probably just looking for a way to sign their JSON and perhaps the name should reflect that. I was wondering if we should go further and have aliases like `"cryptosuite": "json-2023"` instead... 'cause that might be better than `eddsa`.

Reply:
I think it should say `jcs` if JCS is required.

Reply:
-1 to JCS being in the name for at least the following two reasons:

* We don't state the canonicalization algorithm name in any of the other cryptosuites.
* Even if we state it, developers are unlikely to know or care about `jcs`

Is there an alternative name, that will help developers understand that this signature works on "JSON documents", that either of you would be willing to use other than `jcs`?

Reply:
`UrndaEd25519Signature2020.` <- this would be a much more "honest" name...

Callings JCS "JSON" is very misleading.

Reply:
I don't think we're going for "honesty" here... or accuracy. We're going for something that a non-expert developer can easily understand (because they're almost surely not going to be able to understand all the parameters that are in play).

Like: "selective-disclosure-2024" is probably better than "sdjwt-2024" or "bbsdsa-2023".

I'm cringing at the fact that we have "eddsa" vs. "ecdsa" at present... one letter difference (granted, they're both elliptic curves, where one of them is believed to possibly have been tampered with by the NSA and the other one is provably untampered with relation to the curve parameters).

Perhaps "nist-2019" would be better?

I'll note that even the HPKE folks picked something like `HPKE-V1-BASE` or something, somewhere in their specs.

Reply:
> I'm cringing at the fact that we have "eddsa" vs. "ecdsa" at present.

So am I... what happens if you need to add support for new crypto?

All this can be hidden in the "suite name" or in a "parameter registry"... or a combo.

Why not use `DigitalSignature2023`, and then in the registry entry for the suite, explain that is EdDSA with URDNA2015 and sha256?

Or name it `DigitalSignature-EdDSA-URNDA2015-SHA-256-2023`

The name doesn't need to expose all the parameters, if you learn all of them from reading its definition in a registry...

The current approach, uses a mixture of the suite name and parameters... If you want to be like HPKE, you would create a register for each parameter class, and then call the signature class something like:

DataIntegrityProof which would define support for all the predicates necessary to represent the registry entries that exist, for example:

```json
"type": "DataIntegrityProof",
"canonicalization": "JCS",
"digest": "SHA-256",
"cipherSuite": "EdDSA",
"proofValueMultibase": "base64 \ base58 \ base32 \ baseEmoji ..."
```

Or you could do this:

```
"type": "DataIntegrityProof2023",
"proofValueMultibase": "base64 \ base58 \ base32 \ baseEmoji ..."
```

And in the registry say.... "DataIntegrityProof2023" is "JCS" + "SHA-256" + "EdDSA"

Reply:
@OR13 wrote:
> And in the registry say.... "DataIntegrityProof2023" is "JCS" + "SHA-256" + "EdDSA"

Yes, exactly -- that's more or less where I hope we'll end up (as a WG).

We use a cryptosuite name that's easy for devs, without any/much security training, to pick and use as they're copy-pasting their way through an implementation (which is the default mode of operation for most developers). We can elaborate what that term means in a directory or the spec.

> > I'm cringing at the fact that we have "eddsa" vs. "ecdsa" at present.
> 
> So am I... what happens if you need to add support for new crypto?

You use a new base name and/or you iterate the version/date?

> All this can be hidden in the "suite name" or in a "parameter registry"... or a combo.

Yep, exactly.

> Why not use `DigitalSignature2023`, and then in the registry entry for the suite, explain that is EdDSA with URDNA2015 and sha256?

Yes, that's exactly what I'm trying to say.

> Or name it `DigitalSignature-EdDSA-URNDA2015-SHA-256-2023`

No, thanks... that's cryptobabble to most developers. 

That and `DigitalSignature-EdDSA-URNDA2015-MD5` would probably look so similar to most devs that mistakes would be made. In fact, that is exactly how mistakes are made today during algorithm selection with TLS <1.3 and `alg=none` and using `HS256` instead of `ES256`. 

We've gotta stop enabling devs to pick dangerous parameters.

> The name doesn't need to expose all the parameters, if you learn all of them from reading its definition in a registry...

Yes, exactly.

> The current approach, uses a mixture of the suite name and parameters... 

How so? The current (primary) trajectory we're on is: `ecdsa-2019`, `eddsa-2022`, and `bbs-2023` -- none of those let devs pick parameters. `jcs` would be the first time where we introduce this... and, arguably `json` isn't a whole lot better. A better name could be something like `"cryptosuite": "simple-json-2023"`.

> DataIntegrityProof which would define support for all the predicates necessary to represent the registry entries that exist, for example:
> 
> ```json
> "type": "DataIntegrityProof",
> "canonicalization": "JCS",
> "digest": "SHA-256",
> "cipherSuite": "EdDSA",
> "proofValueMultibase": "base64 \ base58 \ base32 \ baseEmoji ..."
> ```

The above suggests that developers can pick and chose the parameters, which we don't want them to be able to do.

> Or you could do this:
> 
> ```json
> "type": "DataIntegrityProof2023",
> "proofValueMultibase": "base64 \ base58 \ base32 \ baseEmoji ..."
> ```

This would put us back into the "cryptosuite type proliferation" that you argued strongly against, and is why we went to `DataIntegrityProof` with a `cryptosuite` parameter in the first place.

That said, feels like we're converging on an approach. Given that we want to support EdDSA, ECDSA, some kind of selective disclosure (in time), and some kind of unlinkable signature (in time)... and given that the parameters for those algorithms are going to change over time... whats a unified naming scheme we can apply to these cryptosuite names?